### PR TITLE
[Model Monitoring] Remove drift analysis from `get_or_create_model_endpoint()`

### PR DESCRIPTION
**Pull request by Eyal-Danieli**
After exposing the new `get_or_create_model_endpoint()` in MLRun SDK (https://github.com/mlrun/mlrun/pull/4020), in this PR we would like to remove the drift analysis process from `get_or_create_model_endpoint()`.
As for the batch inference job, instead of calling `get_or_create_model_endpoint()`, it will use a new SDK function called `record_results()` which wraps both `get_or_create_model_endpoint()` function and the drift analysis process.

A related JIRA ticket: https://jira.iguazeng.com/browse/ML-4443

### CODE REVIEW

**Review by davesh0812 — APPROVED** [review]
LGTM

File: mlrun/common/schemas/model_monitoring/constants.py [hunk 1]
```diff
@@ -74,6 +74,8 @@ class EventFieldType:
     FEATURE_SET_URI = "monitoring_feature_set_uri"
     ALGORITHM = "algorithm"
     VALUE = "value"
+    DRIFT_DETECTED = "drift_detected"
+    POSSIBLE_DRIFT = "possible_drift"
```
[the remark this answers is not in the document]
Comment (Eyal-Danieli) [reply mlrun/common/schemas/model_monitoring/constants.py]: https://github.com/mlrun/mlrun/pull/4135/commits/9c3a0f7a47740e00f4513f2ac966b61a677d0a05

File: mlrun/model_monitoring/api.py [hunk 2]
```diff
@@ -39,117 +39,193 @@
...
+    :param model_path:               The model Store path.
+    :param model_endpoint_name:      If a new model endpoint is generated, the model endpoint name will be presented
+                                     under this endpoint.
+    :param endpoint_id:              Model endpoint unique ID. If not exist in DB, will generate a new record based
+                                     on the provided `endpoint_id`.
+    :param function_name:            If a new model endpoint is created, use this function name for generating the
+                                     function URI.
+    :param context:                  MLRun context. Note that the context is required for logging the artifacts
+                                     following the batch drift job.
+    :param df_to_target:             DataFrame that will be stored under the model endpoint parquet target. This
```
Comment (assaf758) [inline mlrun/model_monitoring/api.py:172]: 1. any assumptions on the structure of this df? Should be stated.
2. cosider change of param name - "infer_results_df"? 
3. "DataFrame will be used by the scheduled monitoring batch drift job" - too much implementation info for user-facing funciton. how about "Will be used for doing the drift analysis"

Comment (Eyal-Danieli) [reply mlrun/model_monitoring/api.py:172]: 1. yes, it should include the inputs + output columns as expected by the batch job (corresponding to the reference data from the training process). At the moment I'll explain in the docs but in the future we can add here some kind of validation engine on that df with error handling (e.g. what to do with `None` cells). 
2. good suggestion. will change it. 
3. agreed :)

**Review by assaf758 — APPROVED** [review]

**Pull request merged**